lib/middleware.js

```diff
@@ -252,5 +252,5 @@ function compare(pathA, pathB) {
     overlap.push(pathA.pop());
     pathB.shift();
   }
-  return overlap.join(sep);
+  return overlap.join('/');
```

Comment:
we want to compare to the `path`, which is a url path, so we should reconstruct the overlap as a url path, indicating we should use `/`